### PR TITLE
Fixes for SRV records handling in hostdb rewrite

### PR DESCRIPTION
**Pull request by cmcfarlen**
Production testing uncovered bugs in SRV record handling.

### CODE REVIEW

File: src/iocore/hostdb/HostDBInfo.cc [hunk 1]
```diff
@@ -73,7 +73,7 @@ HostDBInfo::assign(SRV const *srv, char const *name) -> self_type &
   data.srv.srv_priority = srv->priority;
   data.srv.srv_port     = srv->port;
   data.srv.key          = srv->key;
-  data.srv.srv_offset   = reinterpret_cast<char const *>(this) - name;
+  data.srv.srv_offset   = name - reinterpret_cast<char const *>(this);
```
Comment (bneradt) [inline src/iocore/hostdb/HostDBInfo.cc:76]: Now that you figured this out, can you please add a comment explaining this pointer arithmetic? Or is there a comment elsewhere? The calling logic which loops over `rr_info` is not trivial for me to think through.

**Review by bneradt — CHANGES_REQUESTED** [review]

File: src/iocore/hostdb/HostDBInfo.cc (hunk 1 above)
Comment (cmcfarlen) [reply src/iocore/hostdb/HostDBInfo.cc:76]: I'll add a comment.  The bigger issue here is that the code assumes that the pointer parameter passed in is one that can be recreated from a 16-bit offset value.  You couldn't, for example, pass in a pointer into the stack.  This is a huge design flaw.  While this change corrects the math, it doesn't address the underlying issue.

File: src/iocore/hostdb/HostDBInfo.cc [hunk 2]
```diff
@@ -65,6 +65,17 @@ HostDBInfo::assign(IpAddr const &addr) -> self_type &
   return *this;
 }
 
+/** Assign SRV record and name
+ *
+ * @param srv Pointer to SRV to assign.
+ * @param name Pointer to hostname for the record
```
Comment (bneradt) [inline src/iocore/hostdb/HostDBInfo.cc:71]: nit: Can you please make these both `@param[in]` to show they are input variables. Along these lines, I suggest: `Pointer to the SRV from which to assign.`

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks great. I just have a small request for a tweak in the doxygen comment.

**Review by bneradt — APPROVED** [review]

File: src/iocore/hostdb/HostDBInfo.cc (hunk 2 above)
Comment (bneradt) [reply src/iocore/hostdb/HostDBInfo.cc]: Thank you!

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**